### DISCUSSION TITLE
Clarification on time-varying Cox formula

### DISCUSSION THREAD

**Discussion by CamDavidsonPilon** [CamDavidsonPilon/lifelines#1191] [Q&A]
@daphnezht writes:

> Quick question, what does the Xi bar denote in this formula? Is that the individual average across all time? Does that influence the estimating result (coefficients)? https://lifelines.readthedocs.io/en/latest/Time%20varying%20survival%20regression.html

> Also wondering does anyone know the difference (if any) between the implementation of R package coxph(Surv(tstart, tstop, event) and CoxTimeVaryingFitter in the lifeline package.

**Comment by CamDavidsonPilon** [comment]
`Xi bar` is the mean of the matrix `X` - wait - there should _not_ be a subscript `i` on it! I'll fix that. 

w.r.t. the differences between lifelines and survival in R: computationally they are the same (i.e. some likelihood, and will produce near-identical coefficients), however survival has more time-varying support and is more feature rich. There's no comprehensive lists of differences, however.

**Reply by daphnezht** [reply]
Thank you for your quick response. Very helpful. So just to confirm that having X bar as part of the formula will not affect the coefficient right? 
Another question is does the package has a function that could help plot the hazard function over time (given one subject's X over time)?

**Reply by CamDavidsonPilon** [reply]
Xbar doesn't not effect the coefficients, no. Ex: 

```python
from lifelines.datasets import load_rossi

rossi = load_rossi()

CoxPHFitter().fit(rossi, 'week', 'arrest').summary.loc['age']

# add 10 to "age"
CoxPHFitter().fit(rossi.assign(age=lambda df: df['age'] + 10), 'week', 'arrest').summary.loc['age']
```

Your second question: 

Not really, because there is an epistemological barrier: if I have a time series of the subject, then I know they must not be dead at each time point of that series, so the hazard is 0.

**Reply by daphnezht** [reply]
Got it. The use case we have (and please let me know if we are using this correctly) is we are using this package/ survival analysis framework to build multi-touch attribution model. The user could convert multiple times.
We set X as the count of marketing channel engagement the users had within the past X days as of the stop_col. We set the event_col as the number conversions they have on the stop_col date.
Therefore, we interpret the hazard function as the intensity of users' potential conversion. That's why we are interested in plotting the graph and visualize users' marketing channel engagement and 'conversion potential' over time.

**Reply by CamDavidsonPilon** [reply]
Maybe - I don't think I'm familiar enough with the domain to help. But the time-varying hazard would just be the exponential of (coefficients (`ctv.params_`) times the demeaned covariate matrix, `X`). Even simpler is to forget about the exponential part, and just focus on the _log_ hazard, which has the same information.

**Reply by MarkPundurs** [reply]
Should there also not be a subscript on Xi bar at https://lifelines.readthedocs.io/en/latest/Survival%20Regression.html#cox-s-proportional-hazard-model ?

**Reply by CamDavidsonPilon** [reply]
yes! Thanks for pointing that out, too

**Reply by Cryptojoyz** [reply]
I have a question, why should we subtract the mean value in the Cox time varying formula?

![image](https://github.com/CamDavidsonPilon/lifelines/assets/118330059/42c4490c-7d63-4a54-8d21-5a2a068f6b93)

For example, in this example, the actual mean value of var1 should be 0.3, because var1 is a variable that does not change over time, the sample with ID1 actually has only one var1 value, even though the sample with ID 1 has three var1 values. But when calculating the matrix mean this value becomes 0.2. What is the actual meaning of the matrix mean?
